lib/wickedbase.pm

```diff
@@ -235,7 +236,15 @@ sub ping_with_timeout {
             if ($args{count_success} > 1) {
                 my $cnt = $args{count_success} - 1;
                 $ping_command =~ s/\s-c\s1\s+/ -c $cnt /;
-                validate_script_output($ping_command, sub { /\s+0% packet loss/ });
+                my $ping_out = script_output($ping_command, proceed_on_failure => 1);
+                $ping_out =~ /, (\d{1,3})% packet loss/;
+                #we treat interface in workable state if it manage to echo more than half packets
+                if ($1 > $args{threshold}) {
+                    die('PING EXCEED THRESHOLD ' . $args{threshold} . '%\n' . $ping_out);
+                }
+                elsif (!$1) {
+                    record_info('WARNING', 'PING with packed loss ' . $args{threshold} . '%\n' . $ping_out);
```

Comment:
```suggestion
                    record_info('WARNING', 'PING with packed loss higher than threshold' . $args{threshold} . '%\n' . $ping_out);
```

Reply:
it is lower than threshold actually if it is higher we dying above ...

Reply:
you are right, then, something like `"WARNING", "PING with 25% packet loss. Threshold is 50%. ... "`